plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/IntegerProjection.java

```diff
@@ -54,7 +54,7 @@ public List<String> getProjectedValues(Optional<Domain> partitionValueFilter)
             int currentValue = current;
             String currentValueFormatted = digits
                     .map(digits -> String.format("%0" + digits + "d", currentValue))
-                    .orElse(Integer.toString(current));
+                    .orElseGet(() -> Integer.toString(currentValue));
```

Comment:
(X) Trivial calculation

Reply:
is int toString really trivial?

Reply:
Keeping this based on the assumption that i "looks better" (to me), and it doesn't matter anyway.